core/unix/src/TUnixSystem.cxx

```diff
@@ -568,6 +573,94 @@ TUnixSystem::~TUnixSystem()
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Initialize Unix system interface.
+extern "C"
+{
+   static int FullWrite(int fd, const char *text)
```

Comment:
Then I propose to call is SignalSafeWrite :)

thanks,
Philippe

On 10/19/15 12:26 PM, Zhe Zhang wrote:
>
> In core/unix/src/TUnixSystem.cxx <https://github.com/root-mirror/root/pull/97#discussion_r42399962>:
>
> > @@ -568,6 +573,94 @@ TUnixSystem::~TUnixSystem()
> >
> >  ////////////////////////////////////////////////////////////////////////////////
> >  /// Initialize Unix system interface.
> > +extern "C"
> > +{
> > +   static int FullWrite(int fd, const char *text)
>
> I referred to Brian's code in cmssw. I guess it is called "FullWrite" is because it has been used in signal handler and we want to make sure it is async-signal-safe (not interrupted in the middle).
>
> —
> Reply to this email directly or view it on GitHub <https://github.com/root-mirror/root/pull/97/files#r42399962>.
>